### PR TITLE
Add net6.0-tizen and net6.0-android support

### LINKED ISSUE #10819: Add support for net6.0-tizen and net6.0-android
We should add support for `net6.0-tizen` as specified [here](https://github.com/dotnet/designs/blob/main/accepted/2021/net6.0-tfms/net6.0-tfms.md#design).

/cc @zkat @nkolev92 @WonyoungChoi

### PR DESCRIPTION
**Pull request by zkat**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10819

Regression? Last working version: No

## Description

This PR adds tizen and monoandroid fallback support to net6.0.

n.b. : Only the top commit is what matters here. This PR requires #4011 to get merged first

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

**Comment by zkat** [comment]
@zivkan @nkolev92 this one should be ready now too :)

File: test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs [hunk 1]
```diff
@@ -15,9 +15,13 @@ public class CompatibilityTests
         [InlineData("net6.0-macos1.0", "xamarin.mac", true)]
         // MacCatalyst is a special fallback
         [InlineData("net6.0-maccatalyst1.0", "xamarin.ios", true)]
+        [InlineData("net6.0-tizen9.0", "tizen9.0", true)]
+        [InlineData("net6.0-android12.0", "monoandroid12.0", true)]
         [InlineData("net6.0-ios1.0", "xamarin.mac", false)]
         [InlineData("net6.0-tvos1.0", "xamarin.mac", false)]
         [InlineData("net6.0-maccatalyst1.0", "xamarin.mac", false)]
+        [InlineData("net6.0-tizen9.0", "xamarin.mac", false)]
```
Comment (nkolev92) [inline test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs:23]: can you please add the `net6.0-monikerThatDoesntExist` tests per https://github.com/NuGet/NuGet.Client/pull/4011#discussion_r628765594

**Review by nkolev92 — COMMENTED** [review]
Given the ordering of these checks, is the perf worse for tizen and android?

**Comment by nkolev92** [comment]
@zkat Can you please create a follow up issue to improve the perf of these checks?

File: test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs (hunk 1 above)
Comment (zkat) [reply test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs:23]: done.

**Comment by zkat** [comment]
wow ok this is frustrating.

While setting up microbenchmarks for the tizen/android case to compare, I noticed something weird:
image

Why would all but one of the tests have the same timing? And why would the xamarin tests suddenly be a lot faster?

So then I decided to try running every benchmark _individually_, and...
image

I'm not sure what's going on, but it sounds like we might not actually have any perf issues after all and the test runner was just being weird? :|

I even cranked it up to 10 million iterations and:
image


This is what I'm using to bench this:
```csharp
        private static void Benchmark(Action act, int iterations)
        {
            GC.Collect();
            act.Invoke(); // run once outside of loop to avoid initialization costs
            Stopwatch sw = Stopwatch.StartNew();
            for (int i = 0; i < iterations; i++)
            {
                act.Invoke();
            }
            sw.Stop();
            Console.WriteLine((sw.ElapsedMilliseconds / iterations).ToString());
        }

        [Theory]
        [InlineData("net6.0-tizen9.0", "tizen9")]
        [InlineData("net6.0-android12.0", "monoandroid")]
        [InlineData("net6.0-macos10.0", "xamarin.mac")]
        [InlineData("net6.0", "net5.0")]
        public void Compatibility_Benchmark(string project, string package)
        {
            var projectFramework = NuGetFramework.Parse(project);
            var packageFramework = NuGetFramework.Parse(package);
            var compat = DefaultCompatibilityProvider.Instance;

            Benchmark(() => compat.IsCompatible(projectFramework, packageFramework), 1000000);
        }
```

**Comment by zivkan** [comment]
This is an example of why you should use BenchmarkDotNet to create benchmarks, rather than rolling your own.

.NET JIT compiles classes on first use, to avoid having a long delay on start up. Therefore, the first test to call `NuGetFramework.Parse` is going to be slow while NuGetFramework, and all the APIs that NuGetFramework.Parse uses, gets JIT compiled. The next test that calls NuGetFramework.Parse does not have this JIT cost.

My memory is that the first test method in the class gets "charged" the time for the class constructor, so you can't even call `NuGetFramework.Parse` in the class constructor one time to try to use unit tests as a rudimentary benchmark.

I wish I remembered this last week.

**Review by JonDouglas — APPROVED** [review]
🚢🐿

**Comment by zkat** [comment]
Here's results for IsCompatible using BenchmarkDotNet:

|       Method |     target |             dep |       Mean |    Error |   StdDev |
|------------- |----------- |---------------- |-----------:|---------:|---------:|
| compat | net6.0-ios |          net6.0 |   697.5 ns |  6.33 ns |  5.92 ns |
| compat | net6.0-ios | xamarin.android |   613.0 ns |  4.48 ns |  4.19 ns |
| compat | net6.0-ios |     xamarin.ios | 1,700.4 ns | 20.08 ns | 18.78 ns |

The difference is less noticeable with FrameworkReducer, but I'm not posting it because the output I got is too ugly/hard to parse.

**Review by dominoFire — APPROVED** [review]
